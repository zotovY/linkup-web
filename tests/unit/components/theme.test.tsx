/// <reference types="@types/jest" />
import { render, screen, ThemeWrapper } from "../../test-utils";
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components'

import ThemeComponent, { Props } from "@/components/theme";
import { cleanup, fireEvent } from "@testing-library/react";


describe("Testing theme component", () => {

    afterEach(cleanup);

    const props: Props = {
        isSelected: false,
        gradient: ["#3A3A60", "#19192F"],
        name: "Dark theme",
    };

    test("should select a theme", () => {
        let isSelected = false;
        const onSelect = () => isSelected = !isSelected;

        // Arrange
        const { rerender, getByText } = render(
                <ThemeComponent
                        { ...props }
                        isSelected={ isSelected }
                        onClick={ onSelect }/>
        );
        const component = screen.getByTestId("theme");
        expect(getByText(props.name)).toHaveStyle("font-weight: normal");

        // Act
        fireEvent.click(component);
        rerender(<ThemeComponent { ...props } isSelected={ isSelected } onClick={ onSelect }/>);

        // Assert
        expect(isSelected).toBe(true);
        expect(getByText(props.name)).toHaveStyle("font-weight: 500");
    });

    test("should renders correctly", () => {
        const tree = renderer
                .create(<ThemeWrapper><ThemeComponent { ...props }/></ThemeWrapper>)
                .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test("should renders correctly as selected", () => {
        const tree = renderer
                .create(<ThemeWrapper><ThemeComponent { ...props } isSelected={ true }/></ThemeWrapper>)

        expect(tree.toJSON()).toMatchSnapshot();
    });
});
