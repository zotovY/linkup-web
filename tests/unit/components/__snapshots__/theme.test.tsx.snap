// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing theme component should renders correctly 1`] = `
<div
  className="theme__Container-sc-1q5y7sm-0 fUCSLI theme-component"
  data-testid="theme"
  name="Dark theme"
>
  <div
    className="theme__Preview-sc-1q5y7sm-1 fdTDcu"
    name="Dark theme"
  />
  <span
    className="theme__Name-sc-1q5y7sm-2 izzcna"
    name="Dark theme"
  >
    Dark theme
  </span>
</div>
`;

exports[`Testing theme component should renders correctly as selected 1`] = `
<div
  className="theme__Container-sc-1q5y7sm-0 fUCSLI theme-component"
  data-testid="theme"
  name="Dark theme"
>
  <div
    className="theme__Preview-sc-1q5y7sm-1 fdTDcu"
    name="Dark theme"
  />
  <span
    className="theme__Name-sc-1q5y7sm-2 gmVQuL"
    name="Dark theme"
  >
    Dark theme
  </span>
</div>
`;
